lib/matplotlib/axes/_base.py

```diff
@@ -2703,11 +2706,17 @@ def ticklabel_format(self, **kwargs):
         useMathText = kwargs.pop('useMathText', None)
         axis = kwargs.pop('axis', 'both').lower()
         if scilimits is not None:
-            try:
-                m, n = scilimits
-                m + n + 1  # check that both are numbers
-            except (ValueError, TypeError):
-                raise ValueError("scilimits must be a sequence of 2 integers")
+            if type(scilimits) == int:
```

Comment:
Doing `type(foo) == XX` tests is not the best way to do this sort of thing (for example, it will fail on `type(np.int8(5)) == int` which is something that a user would reasonably expect to work) and it will fail on all sub-classes.  It is better to do `isinstance(foo, XX)` which will correctly handle sub-classes (but that still fails `np.int8`).  In this case I think something like

```python
try:
   m, n = scilimits
except (TypeError, ValueError):
   try:
      m = n = scilimits
   except TypeError:
       raise ValueError(...)
try:
    m + n + 1
except TypeError:
    raise ValueError(...)
```

It also looks like below `scilimits` is passed through to `set_powerlimits` (not m and n).